pkg/storaged/smart-details.jsx

```diff
@@ -0,0 +1,120 @@
+
+const _ = cockpit.gettext;
+
+const selftestStatusDescription = {
+    success: _("Successful"),
+    aborted: _("Aborted"),
+    interrupted: _("Interrupted"),
+    fatal: _("Did not complete"),
+    error_unknown: _("Failed (Unknown)"),
+    error_electrical: _("Failed (Electrical)"),
+    error_servo: _("Failed (Servo)"),
+    error_read: _("Failed (Read)"),
+    error_handling: _("Failed (Damaged)"),
+    inprogress: _("In progress"),
+};
+
+const SmartActions = ({ smartInfo }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+    const smartSelftestStatus = smartInfo.SmartSelftestStatus;
+
+    const runSmartTest = async (type) => {
+        await smartInfo.SmartSelftestStart(type, {});
+    };
+
+    const abortSmartTest = async () => {
+        await smartInfo.SmartSelftestAbort({});
+    };
+
+    const actions = [
+        <DropdownItem key="run-short-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('short') }}>
+            {_("Run short test")}
+        </DropdownItem>,
+        <DropdownItem key="run-extended-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('extended') }}>
+            {_("Run extended test")}
+        </DropdownItem>,
+        <DropdownItem key="run-conveyance-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('conveyance') }}>
+            {_("Run conveyance test")}
+        </DropdownItem>,
+    ];
+
+    if (smartInfo.SmartSelftestStatus === "inprogress") {
+        actions.push(
+            <DropdownItem key="abort-smart-test"
+                          onClick={() => { setKebabOpen(false); abortSmartTest('conveyance') }}>
+                {_("Abort test")}
+            </DropdownItem>,
+        );
+    }
+
+    return (
+        <Dropdown toggle={<KebabToggle onToggle={(_, isOpen) => setKebabOpen(isOpen)} />}
+                isPlain
+                isOpen={isKebabOpen}
+                position="right"
+                id="smart-actions"
+                dropdownItems={actions} />
+    );
+};
+
+export const SmartDetails = ({ smartInfo }) => {
+    const SmartDetailRow = ({ title, value }) => {
+        if (value === undefined)
+            return null;
+
+        return (
+            <DescriptionListGroup>
+                <DescriptionListTerm>{title}</DescriptionListTerm>
+                <DescriptionListDescription>{value}</DescriptionListDescription>
+            </DescriptionListGroup>
+        );
+    };
+
+    return (
+        <Card>
+            <CardHeader actions={{ actions: <SmartActions smartInfo={smartInfo} /> }}>
+                <CardTitle component="h2">{_("S.M.A.R.T")}</CardTitle>
```

Comment:
I don't think that's really something one can translate?

Reply:
hmm force of habit